From 0d45b2d920333c21aef27a8f7de3e27d44dc23d3 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Mon, 12 Aug 2024 14:46:31 +0800
Subject: [PATCH 28/29] third_party/libvpx: Only build LSX optimizations when
 loongarch64_use_lsx is true.

The LSX optimizations are now unconditionally depended on LA64 (64-bit LoongArch, or LoongArch64) by BUILD.gn.

Update BUILD.gn to depend on them only when loongarch64_use_lsx is true.

Bug: 359039635
Change-Id: I28651f0672fe825812931c254db28682adb41e68

https://chromium-review.googlesource.com/c/chromium/src/+/5781629/3
---
 third_party/libvpx/BUILD.gn | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/third_party/libvpx/BUILD.gn b/third_party/libvpx/BUILD.gn
index ef12741012..e8749b0a80 100644
--- a/third_party/libvpx/BUILD.gn
+++ b/third_party/libvpx/BUILD.gn
@@ -5,6 +5,7 @@
 import("//build/config/android/config.gni")
 import("//build/config/arm.gni")
 import("//build/config/chromeos/ui_mode.gni")
+import("//build/config/loongarch64.gni")
 import("//testing/test.gni")
 import("//third_party/libvpx/libvpx_srcs.gni")
 import("//third_party/nasm/nasm_assemble.gni")
@@ -596,7 +597,7 @@ static_library("libvpx") {
   if (current_cpu == "arm" && arm_assembly_sources != []) {
     deps += [ ":libvpx_assembly_arm" ]
   }
-  if (current_cpu == "loong64") {
+  if (current_cpu == "loong64" && loongarch64_use_lsx) {
     deps += [ ":libvpx_loongarch_lsx" ]
   }
 
-- 
2.46.0

